src/WLPToken.sol

```diff
@@ -19,75 +20,98 @@ import "./interfaces/ILPToken.sol";
  * staking it and wrapping the received lpToken.
  *
  */
-contract WLPToken is ERC20PermitUpgradeable {
+contract WLPToken is ERC4626Upgradeable {
     ILPToken public lpToken;
 
     function initialize(ILPToken _lpToken) public initializer {
-        __ERC20Permit_init("Wrapped lpToken");
-        __ERC20_init("Wrapped lpToken", "wlpToken");
+        __ERC20_init("Wrapped LP Token", "wlpToken");
+        __ERC4626_init(IERC20Upgradeable(address(_lpToken)));
         lpToken = _lpToken;
     }
 
     /**
-     * @notice Exchanges lpToken to wlpToken
-     * @param _lpTokenAmount amount of lpToken to wrap in exchange for wlpToken
-     * @dev Requirements:
-     *  - msg.sender must approve at least `_lpTokenAmount` lpToken to this
-     *    contract.
-     * @return Amount of wlpToken user receives after wrap
+     * @dev Returns the total assets managed by the vault.
+     * Overrides the virtual totalAssets method of ERC4626.
+     * @return The total amount of lpToken held by the vault.
      */
-    function wrap(uint256 _lpTokenAmount) external returns (uint256) {
-        require(_lpTokenAmount > 0, "wlpToken: can't wrap zero lpToken");
-        uint256 _wlpTokenAmount = lpToken.getSharesByPooledEth(_lpTokenAmount);
-        _mint(msg.sender, _wlpTokenAmount);
-        lpToken.transferFrom(msg.sender, address(this), _lpTokenAmount);
-        return _wlpTokenAmount;
+    function totalAssets() public view override returns (uint256) {
+        return lpToken.balanceOf(address(this));
     }
 
     /**
-     * @notice Exchanges wlpToken to lpToken
-     * @param _wlpTokenAmount amount of wlpToken to uwrap in exchange for lpToken
-     * @return Amount of lpToken user receives after unwrap
+     * @dev Converts an amount of lpToken to the equivalent amount of shares.
+     * @param assets Amount of lpToken.
+     * @return The equivalent shares.
      */
-    function unwrap(uint256 _wlpTokenAmount) external returns (uint256) {
-        require(_wlpTokenAmount > 0, "wlpToken: zero amount unwrap not allowed");
-        uint256 _lpTokenAmount = lpToken.getPooledEthByShares(_wlpTokenAmount);
-        _burn(msg.sender, _wlpTokenAmount);
-        lpToken.transfer(msg.sender, _lpTokenAmount);
-        return _lpTokenAmount;
+    function convertToShares(uint256 assets) public view override returns (uint256) {
```

Comment:
its better to override `_convertToShares` and `_convertToAssets` since they are used in other funcs: `maxWithdraw` and preview functions